### PR TITLE
chore: begin adding logger to `zarf dev`

### PR DESCRIPTION
**Pull request by mkcp**
## Description
This PR adds logger to `sha256sum`, `patch-git`, `lint` and `find-images`. There's a few more dev commands to update, which i'll grab in another PR.

## Related Issue
Relates to #2576 

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/zarf-dev/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/pkg/packager/prepare.go [hunk 1]
```diff
@@ -260,6 +269,8 @@ func (p *Packager) findImages(ctx context.Context) (map[string][]string, error)
 			}
 		}
 
+		imgCompStart := time.Now()
+		l.Info("looking for image in component", "name", component.Name, "resourcesCount", len(resources))
```
Comment (AustinAbro321) [inline src/pkg/packager/prepare.go:273]: image->images

File: src/pkg/packager/prepare.go [hunk 2]
```diff
@@ -305,16 +318,23 @@ func (p *Packager) findImages(ctx context.Context) (map[string][]string, error)
 		}
 
 		spinner.Success()
+		l.Debug("done looking for image in component",
```
Comment (AustinAbro321) [inline src/pkg/packager/prepare.go:321]: image->images

File: src/cmd/dev.go [hunk 3]
```diff
@@ -104,20 +105,30 @@ var devTransformGitLinksCmd = &cobra.Command{
...
-		processedText := transform.MutateGitURLsInText(message.Warnf, pkgConfig.InitOpts.GitServer.Address, text, pkgConfig.InitOpts.GitServer.PushUsername)
+
+		// Set log func for transform
+		var logFn func(string, ...any)
+		logFn = message.Warnf
+		// Use logger if we can
+		if logger.Enabled(ctx) {
+			logFn = logger.From(ctx).Warn
+		}
+		processedText := transform.MutateGitURLsInText(logFn, gitServer.Address, text, gitServer.PushUsername)
```
Comment (AustinAbro321) [inline src/cmd/dev.go:131]: This is actually expecting printf style logging so the message ends up looking like this. 
```bash
 Unable to transform the git url, using the original url we have: %s !BADKEY=https://*/*.git
```
You could instead use this to keep with the printf format
```go
		l := logger.From(ctx)
		logFn = slog.NewLogLogger(l.Handler(), slog.LevelWarn).Printf
```
Might be good to add a todo to this function to move to just use the slogger from context after logging is released also. Not sure why this was given it's own logging function. Doesn't looked to be called from anywhere outside of Zarf

File: src/pkg/packager/prepare.go (hunk 1 above)
Comment (mkcp) [reply src/pkg/packager/prepare.go:273]: Good catch, ty

File: src/cmd/dev.go (hunk 3 above)
Comment (mkcp) [reply src/cmd/dev.go:131]: Yeah that's bugged for sure -- I'm going to tear it out and make a todo for later. I can see the effort to make a logger you can pass in, but it just creates weird compatibility issues. the Printf on log.Logger is neat though, hadn't seen that before.

**Review by AustinAbro321 — APPROVED** [review]

**Pull request merged**